cpp/include/cudf/transform.hpp

```diff
@@ -83,6 +86,36 @@ std::pair<std::unique_ptr<rmm::device_buffer>, size_type> nans_to_nulls(
 std::pair<std::unique_ptr<rmm::device_buffer>, cudf::size_type> bools_to_mask(
   column_view const& input, rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
 
+/**
+ * @brief Create `arrow::Table` from cudf table `input`
+ *
+ * Converts the `cudf::table_view` to `arrow::Table` with the provided
+ * metadata `column_names`.
+ *
+ * @throws cudf::logic_error if `column_names` size doesn't match with number of columns.
+ *
+ * @param input table_view that needs to be converted to arrow Table
+ * @param column_names Vector of column names for metadata of arrow Table
+ * @param ar_mr arrow memory pool to allocate memory for arrow Table
+ * @return arrow Table generated from given input cudf table
+ **/
+std::shared_ptr<arrow::Table> cudf_to_arrow(
```

Comment:
```suggestion
std::unique_ptr<arrow::Table> cudf_to_arrow(
```

Reply:
arrow::Table can be created with only arrow::Table::Make which returns a shared pointer, and moving a shared pointer to a unique pointer doesn't feel right to me.

Reply:
Ah, well then that's Arrow's fault for returning `shared_ptr` by default. I guess there's nothing we can do about it.